libsupport/include/katana/OpaqueID.h

```diff
@@ -46,6 +46,20 @@ struct OpaqueID {
   //  OpaqueID<_IDType, _Value>. There doesn't seem to be any way to do that.
   using ValueType = _Value;
 
+  // Default count type is size_t
+  template <class T, class = void>
+  struct integral_traits {
+    using Count = size_t;
+  };
+
+  // If value type is integral, use the unsigned version of it for Count
+  template <class T>
+  struct integral_traits<T, typename std::enable_if_t<std::is_integral_v<T>>> {
+    using Count = std::make_unsigned_t<T>;
+  };
```

Comment:
I would rather call this `count_traits`. `integral_traits` is too similar to to the names of the trait classes in STL.

Also I would rather have this type in the `katana` namespace.

Reply:
Ok, easy to change the name to `count_traits`.

Can you explain your namespace comment?  `Count `is a type in `OpaqueID`, which is in the `katana `namespace.

Reply:
The reason for putting it in the namespace is mostly convention: Traits classes go in the the namespace. However the convention is there for a reason. If someone needs to specialize your trait for some custom type, they can. If it's in a class, they cannot (assuming I understand C++ correctly, which is always in question). If you disagree with this argument, go ahead and ignore this request. I don't have a strong opinion.

Reply:
Arthur, I wrote this code by reading a bunch of tutorials and stackoverflow pages and bribing the monkeys in my brain to keep typing by feeding them bananas for hours.  My questions arise from ignorance, not obstinance.

I pulled the `count_traits` code out of `OpaqueID`.  Is that what you wanted?

Reply:
hehehe. I know how that goes. That is what I wanted. And I wasn't trying to imply obstinance. :-) Just that I don't feel like my argument is sound enough to override other opinions. But if you want no other opinions. So be it. :-)